src/program/latex/mod.rs

```diff
@@ -0,0 +1,1600 @@
+    Ctrl(String),
+    /// `\targ{}`: Make a controlled-not gate.
+    Targ,
+    /// `\control{}`: Make a controlled-phase gate--different from Ctrl.
+    Control,
+    /// `\swap{wire}`: Make a swap gate--used with TargX.
+    Swap(String),
+    /// `\targX{}`: Make a qubit the target for a swap--used with Swap.
+    TargX,
+}
+
+impl Command {
+    /// Returns the LaTeX String for a given Command variant.
+    ///
+    /// # Arguments
+    /// `command` - A Command variant.
+    ///
+    /// # Examples
+    /// ```
+    /// use quil_rs::program::latex::Command;
+    /// let ket_0 = "0".to_string();
+    /// let lstick_ket_0 = Command::get_command(Command::Lstick(ket_0));
+    /// ```
+    pub fn get_command(command: Self) -> String {
+        match command {
+            Self::Lstick(wire) => format(format_args!(r#"\lstick{{\ket{{q_{{{wire}}}}}}}"#)),
+            Self::Rstick(wire) => format(format_args!(r#"\rstick{{\ket{{q_{{{wire}}}}}}}"#)),
+            Self::Gate(name) => format(format_args!(r#"\gate{{{name}}}"#)),
+            Self::Phase(symbol) => format(format_args!(r#"\phase{{{symbol}}}"#)),
+            Self::Super(script) => format(format_args!(r#"^{{\{script}}}"#)),
+            Self::Qw => r"\qw".to_string(),
+            Self::Nr => r"\\".to_string(),
+            Self::Meter(wire) => format(format_args!(r#"\meter{{{wire}}}"#)),
+            Self::Ctrl(wire) => format(format_args!(r#"\ctrl{{{wire}}}"#)),
+            Self::Targ => r"\targ{}".to_string(),
+            Self::Control => r"\control{}".to_string(),
+            Self::Swap(wire) => format(format_args!(r#"\swap{{{wire}}}"#)),
+            Self::TargX => r"\targX{}".to_string(),
+        }
+    }
+}
+
+/// Types of parameters passed to commands.
+#[derive(Debug)]
+pub enum Parameter {
+    /// Symbolic parameters
+    Symbol(Symbol),
+}
+
+impl Clone for Parameter {
+    fn clone(&self) -> Parameter {
+        match self {
+            Parameter::Symbol(symbol) => match symbol {
+                Symbol::Alpha => Parameter::Symbol(Symbol::Alpha),
+                Symbol::Beta => Parameter::Symbol(Symbol::Beta),
+                Symbol::Gamma => Parameter::Symbol(Symbol::Gamma),
+                Symbol::Phi => Parameter::Symbol(Symbol::Phi),
+                Symbol::Pi => Parameter::Symbol(Symbol::Pi),
+                Symbol::Text(text) => Parameter::Symbol(Symbol::Text(text.to_string())),
+            },
+        }
+    }
+}
+
+/// Supported Greek and alphanumeric symbols.
+#[derive(Debug)]
+pub enum Symbol {
+    Alpha,
+    Beta,
+    Gamma,
+    Phi,
+    Pi,
+    Text(String),
+}
+
+impl Symbol {
+    /// Returns the supported Symbol variant from text otherwise stores the
+    /// unsupported symbol as text in the Text variant.
+    ///
+    /// # Arguments
+    /// `text` - a String representing a greek or alaphanumeric symbol
+    pub fn match_symbol(text: String) -> Symbol {
+        if text == "alpha" {
```

Comment:
This is more typically expressed as a `match`.

Reply:
@hiyaryan you resolved the conversation, but it still appears to be an if/else chain rather than a `match`.

Reply:
Maybe you didn't push the changes